akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -61,6 +61,24 @@ abstract class GraphStage[S <: Shape] extends GraphStageWithMaterializedValue[S,
   def createLogic: GraphStageLogic
 }
 
+/**
+ * A SourceStage represents a reusable graph stream processing stage. A SourceStage consists of a [[akka.stream.Shape]] which describes
+ * its output port.
+ */
+abstract class SourceStage[Out, M](name: String) extends GraphStageWithMaterializedValue[SourceShape[Out], M] {
```

Comment:
@drewhk, I tried it in few different ways and the most simple was:
```scala
abstract class SourceStage[Out, M](name: String) extends GraphStageWithMaterializedValue[SourceShape[Out], M] {
  def push(elem:Out) = createLogicAndMaterializedValue._1.push(out,elem)
```
which is calling createLogicAndMaterializedValue and creating GraphStageLogic another time. As far as I understand this call must be made only by materializer. Can we skip this for now (I don't see easy way of implementing this)?